tools/c7n_oci/c7n_oci/provider.py

```diff
@@ -19,15 +21,90 @@ class OCI(Provider):
     resource_prefix = "oci"
     resources = PluginRegistry("%s.resources" % resource_prefix)
     resource_map = ResourceMap
+    _oci_profile_session = None
 
     def initialize(self, options):
+        if options["account_id"] is None:
+            options["oci_config"] = self._get_oci_profile_session(options).get_config()
         return options
 
+    def _get_subscribed_regions(self, options):
+        client = self._get_oci_profile_session(options).client("oci.identity.IdentityClient")
+        subscribed_regions = client.list_region_subscriptions(
+            tenancy_id=options["oci_config"]["tenancy"]
+        ).data
+        regions = [region.region_name for region in subscribed_regions if region.status == "READY"]
+        return regions
+
+    @staticmethod
+    def _is_subscribed_region(subscribed_regions, region):
+        if region in subscribed_regions:
+            return True
+
+        log.info(
+            (
+                "User's tenancy is not subscribed to the %s region. "
+                "So skipping this region for the policy execution."
+            ),
+            region,
+        )
+        return False
+
+    def _get_oci_profile_session(self, options):
+        if self._oci_profile_session is None:
+            self._oci_profile_session = self.get_session_factory(options)()
+        return self._oci_profile_session
+
+    @staticmethod
+    def _join_output(output_dir, suffix):
+        if "{region}" in output_dir:
```

Comment:
we probably need to extract a utility function to get rid of the duplicate implementation

Reply:
Moved this logic to utils.py.

Reply:
Done